sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -606,301 +576,439 @@ abstract class FileParquetPartitionReaderBase(
       out.close()
     }
   }
+
+  protected def readPartFile(
+      blocks: Seq[BlockMetaData],
+      clippedSchema: MessageType,
+      filePath: Path): (HostMemoryBuffer, Long) = {
+    withResource(new NvtxWithMetrics("Buffer file split", NvtxColor.YELLOW,
+      metrics("bufferTime"))) { _ =>
+      withResource(filePath.getFileSystem(conf).open(filePath)) { in =>
+        var succeeded = false
+        val estTotalSize = calculateParquetOutputSize(blocks, clippedSchema)
+        val hmb =
+          HostMemoryBuffer.allocate(estTotalSize)
```

Comment:
Nit: Could use `closeOnExcept` here rather than an explicit `succeeded` boolean var.

Reply:
sure I'll change this was just moved from below